### ISSUE TITLE
[Help]: Version 1.3.1 : Symbol not found: _OBJC_CLASS_$_FlutterError

### ISSUE THREAD

**Issue by elmar001** [boskokg/flutter_blue_plus#703]
### Requirements

- [X] I've looked at the README 'Common Problems' section

### Have you checked this problem on the example app?

No

### FlutterBluePlus Version

1.3.1

### Flutter Version

2.10.5

### What OS?

iOS

### OS Version

iOS17

### Bluetooth Module

Not sure, tested with several iPhones

### What is your problem?

When app is run in Xcode it stucks on the startup screen. If the project is released to TestFlight it will crash on startup. I have narrowed down the issue and it seem to be directly related to Flutter Blue Plus. Removing that package solves the issue. It is big sized Flutter project and upgrading is not relevant at this stage. I wonder why it stopped working now? Note: Not related to iOS 17, already tested with older iOS 16.0. versions.

### Logs

```shell
Xcode runtime error:
dyld`:
    0x1b6bbec24 <+0>:  mov    x16, #0x209
    0x1b6bbec28 <+4>:  svc    #0x80
->  0x1b6bbec2c <+8>:  b.lo   0x1b6bbec48               ; <+36>
    0x1b6bbec30 <+12>: stp    x29, x30, [sp, #-0x10]!
    0x1b6bbec34 <+16>: mov    x29, sp
    0x1b6bbec38 <+20>: bl     0x1b6b62de4               ; cerror_nocancel
    0x1b6bbec3c <+24>: mov    sp, x29
    0x1b6bbec40 <+28>: ldp    x29, x30, [sp], #0x10
    0x1b6bbec44 <+32>: ret    
    0x1b6bbec48 <+36>: ret    
Thread 1: signal SIGABRT



dyld[2347]: Symbol not found: _OBJC_CLASS_$_FlutterError
  Referenced from: <8AE94F1E-06C4-3C99-9A44-090D6815742D> /private/var/containers/Bundle/Application/25DB5D90-EDA4-4795-A29B-9A207F7F35E3/Runner.app/Runner
  Expected in:     <7E6D03A8-D9E7-3784-A3D5-FDC3005A1E64> /private/var/containers/Bundle/Application/25DB5D90-EDA4-4795-A29B-9A207F7F35E3/Runner.app/Frameworks/flutter_blue_plus.framework/flutter_blue_plus
```

**Comment by chipweinberger** [comment]
i've seen this error before. upgrading fixes it.

**Comment by elmar001** [comment]
Thanks @chipweinberger, seems like it relates to Xcode 15 MacOS Sonoma update. Someone already discussed here: https://stackoverflow.com/questions/77144171/dyld2179-symbol-not-found-objc-class-fluttererror
It only affects Flutter Blue Plus, not any other package.

**Comment by chipweinberger** [comment]
same issue https://github.com/boskokg/flutter_blue_plus/issues/581

**Comment by chipweinberger** [comment]
a simple fix is to just fork 1.3.1 and remove the try catch block using FlutterError. this was invalid code. It should be NSException. see master branch for details. i.e.

```
    @catch (NSException *e)
    {
        NSString *stackTrace = [[e callStackSymbols] componentsJoinedByString:@"\n"];
        NSDictionary *details = @{@"stackTrace": stackTrace};
        result([FlutterError errorWithCode:@"iosException" message:[e reason] details:details]);
    }
```
